[CIT-300] Update doc for tedge connect <cloud> --test (#220)
thin-edge/thin-edge.io: docs/src/howto-guides/007_test_connectiom.md

# How to test the connection to cloud?

We provide a way to test the connection from your device to a cloud provider.
You can call this connection check function by

```shell
sudo tedge connect <cloud> --test
```

It returns exit code 0 if the connection check is successful, otherwise, 1.

> Also, this test is a part of the `tedge connect <cloud>` process.

Reviewer: ```suggestion
This test is already performed as part of the `tedge connect <cloud>` command
```

## What does the test do?

> 

Reviewer: I feel we can just add a few lines on what the `test` command does in a generic manner: Send a message to the cloud and wait for a response. The subsequent sections can detail out the cloud specific behaviour, if needed.

> ### For Cumulocity IoT

Reviewer: This section feels like documenting the internal implementation details. Do we really want to expose this to the end user?

Author: Good point. I am not sure. @itsyitsy Can you decide if we should detail out or not?

The test publishes [a SmartREST2.0 static template message for device creation `100`](https://cumulocity.com/guides/device-sdk/mqtt/#a-nameinventory-templatesinventory-templates-1xxa) to the topic `c8y/s/us`.
If the device-twin is already created in your Cumulocity,
the device is supposed to receive `41,100,Device already existing` on the error topic `c8y/s/e`.

So, the test subscribes to `c8y/s/e` topic and if it receives the expected message on the topic, the test is marked successful.

The connection test sends maximum two of SmartREST2.0 `100` requests.
This is because the first `100` request can be considered a successful device creation request if the device-twin does not exist in Cumulocity yet.

### For Azure IoT Hub

> The test subscribes to the topic `az/twin/res/`.

Reviewer: Same comment as above.

# How to test the connection to cloud?

We provide a way to test the connection from your device to a cloud provider.
You can call this connection check function by

```shell
sudo tedge connect <cloud> --test
```

It returns exit code 0 if the connection check is successful, otherwise, 1.

This test is already performed as part of the `tedge connect <cloud>` command.

## What does the test do?

> The connection test sends a message to the cloud and waits for a response.

Reviewer: A line break before this line separating it from the title would be good.
